.github/workflows/checks.yml

```diff
@@ -0,0 +1,98 @@
+# This workflow performs basic checks:
+#
+#   1. run a preparation step to install and cache node modules
+#   2. once prep succeeds, lint and test run in parallel
+#
+# The checks are skipped on the 'main' branch. The project relies on branch
+# protection to avoid pushes straight to 'main'.
+
+name: Checks
+
+on:
+  push:
+    branches-ignore:
+    - 'main'
+
+env:
+  NODE: 14
+
+jobs:
+  prep:
+    if: github.event.pull_request.draft == false
```

Comment:
@olafveerman this rule is unefective if `on:` clause is not targeting PRs. Checks will run on every push not in main branch. In LULC we are using the following rule in checks.yml:

```
on:
  pull_request:
    types:
      - opened
      - synchronize
      - reopened
      - ready_for_review
```

I have tested and it will only run check in non-drafts PRs. Link to LULC:

https://github.com/developmentseed/lulc-frontend/tree/develop/.github/workflows

Reply:
This is a great addition :)